### PR TITLE
fix(state): hard crash when user_id is None

### PR DESCRIPTION
**Pull request by Miolus**
Reflect https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object

## Summary
Fix issue like detailed in https://discord.com/channels/881118111967883295/1084474550345023638

## Checklist

- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/state.py [hunk 1]
```diff
@@ -2307,7 +2307,7 @@ def parse_auto_moderation_action_execution(self, data) -> None:
 
     def parse_guild_audit_log_entry_create(self, data) -> None:
         guild = self._get_guild(int(data["guild_id"]))
-        user_id = int(data["user_id"])
+        user_id = int(data["user_id"]) if data["user_id"] != None else data["user_id"]
```
Comment (toolifelesstocode) [inline nextcord/state.py:2310]:
```suggestion
        user_id = int(data["user_id"]) if data["user_id"] is not None else data["user_id"]
```
As `is`/`is not` is used when making comparisons to classes (`None` in this case)

**Review by toolifelesstocode — CHANGES_REQUESTED** [review]

**Review by toolifelesstocode — APPROVED** [review]
Lgtm if it's been tested

File: nextcord/state.py [hunk 2]
```diff
@@ -2307,7 +2307,7 @@ def parse_auto_moderation_action_execution(self, data) -> None:
 
     def parse_guild_audit_log_entry_create(self, data) -> None:
         guild = self._get_guild(int(data["guild_id"]))
-        user_id = int(data["user_id"])
+        user_id = int(data["user_id"]) if data["user_id"] is not None else data["user_id"]
```
Comment (Skelmis) [inline nextcord/state.py:2310]: Does this possibly want `if data.get("user_id") is not None` to account for when it's not actually present in the data payload?

Comment (EmreTech) [reply nextcord/state.py:2310]: According to the Discord API Docs, it's always in the payload. Do you know of situations where it's not present?

Comment (Skelmis) [reply nextcord/state.py:2310]: No, but given it's nullable I wouldn't put it past discord to just simply not provide in the future

Comment (EmreTech) [reply nextcord/state.py:2310]: Good point.

**Comment by ooliver1** [comment]
In the future I would recommend using a separate branch in your fork other than `master`.

**Review by ooliver1 — APPROVED** [review]

File: nextcord/state.py [hunk 3]
```diff
@@ -2307,13 +2307,15 @@ def parse_auto_moderation_action_execution(self, data) -> None:
 
     def parse_guild_audit_log_entry_create(self, data) -> None:
         guild = self._get_guild(int(data["guild_id"]))
-        user_id = int(data["user_id"])
+        user_id = None if data.get("user_id") is None else int(data["user_id"])
         user = self.get_user(user_id)
+        if user_id is None or user is None:
```
Comment (ooliver1) [inline nextcord/state.py:2312]: Little note on my changes here. The `users` parameter is used as a dictionary to resolve users found inside the entry. I am not sure why it does not use `get_user` in there, but thats irrelevant.

The ternary above is shorter than what you would think would be used.
```py
user_id = None if data.get("user_id") is None else int(data["user_id"])
user_id = int(data["user_id"]) if data.get("user_id") is not None else None
```

**Review by alentoghostflame — APPROVED** [review]
Haven't tested, but looks good to me. This properly reflects ddocs listing `user_id` as a nullable field.

**Comment by shahriyardx** [comment]
After this there is a new error 

```py
Traceback (most recent call last):
  File "/myBot/bot.py", line 214, in <module>
    bot.run(os.environ["TOKEN"])
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/client.py", line 871, in run
    return future.result()
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/client.py", line 850, in runner
    await self.start(*args, **kwargs)
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/client.py", line 814, in start
    await self.connect(reconnect=reconnect)
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/shard.py", line 511, in connect
    raise item.error
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/shard.py", line 152, in worker
    await self.ws.poll_event()
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/gateway.py", line 603, in poll_event
    await self.received_message(msg.data)
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/gateway.py", line 553, in received_message
    func(data)
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/state.py", line 2318, in parse_guild_audit_log_entry_create
    entry = AuditLogEntry(auto_moderation_rules={}, users=users, data=data, guild=guild)
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/audit_logs.py", line 457, in __init__
    self._from_data(data)
  File "/myBot/venv/lib/python3.8/site-packages/nextcord/audit_logs.py", line 519, in _from_data
    channel_id = int(self.extra["channel_id"])
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

**Pull request merged**